lib/hl7/field_grammar.ex

```diff
@@ -1,41 +1,70 @@
 defmodule HL7.FieldGrammar do
   require Logger
 
-  @moduledoc false
+  @derive Inspect
+
+  defstruct [:data]
+
+  @typep segment :: String.t()
+  @typep indices :: [integer, ...]
+  @type t :: %__MODULE__{
+          data: {segment, indices} | indices
+        }
+
+  def to_string(%__MODULE__{data: {segment, indices}}) do
+    "#{segment}-" <> Enum.join(indices, ".")
+  end
 
-  @spec to_indices(String.t()) :: list()
+  def to_string(%__MODULE__{data: indices}) when is_list(indices) do
+    Enum.join(indices, ".")
+  end
+
+  defimpl String.Chars do
+    def to_string(grammar) do
+      HL7.FieldGrammar.to_string(grammar)
+    end
+  end
+
+  @moduledoc false
+  @spec to_indices(String.t()) :: t()
```

Comment:
While we're at it, maybe rename this function to some like `new` or `to_data` since it generates the full struct data and not just numeric indices?